v2/federation/fetch.go

```diff
@@ -61,8 +88,22 @@ func FetchBundle(ctx context.Context, trustDomain spiffeid.TrustDomain, url stri
 	return bundle, nil
 }
 
-type fetchOption func(*fetchOptions)
+type fetchOption func(*fetchOptions) error
+
+func (fo fetchOption) apply(opts *fetchOptions) error {
+	return fo(opts)
+}
+
+type authMethod int
+
+const (
+	authMethodDefault authMethod = iota
+	authMethodSPIFFE
+	authMethodWebPKI
+)
+
+var authMethodName = [3]string{"Default", "SPIFFE", "WebPKI"}
 
-func (fo fetchOption) apply(opts *fetchOptions) {
-	fo(opts)
+func (a authMethod) String() string {
+	return authMethodName[a]
```

Comment:
nit: this pattern isn't very robust.... if you miss adding a string you get a panic and if you add one in the wrong order, it throws off all of the strings in a hard-to-discover way (without exhaustive unit-tests). I'd prefer a switch statement here. The other advantage of using a switch statement is that it could be easy to check for exhaustiveness by a linter (I don't think our current linters do that though).

That being said, if you go with my suggestion on the error message, this function can go away.

Reply:
This function will go away.